release/internal/version/version.go

```diff
@@ -24,52 +50,114 @@ func (v *Version) String() string {
 
 // FormattedString returns the formatted string representation of the version.
 func (v *Version) FormattedString() string {
-	ver := v.String()
-	return fmt.Sprintf("v%v", ver)
+	return fmt.Sprintf("v%v", v.String())
 }
 
-// Milestone returns the milestone of the version.
+// Milestone returns the GitHub milestone name which corresponds with this version.
 func (v *Version) Milestone() string {
 	ver := semver.MustParse(string(*v))
 	return fmt.Sprintf("%s v%d.%d.%d", utils.DisplayProductName(), ver.Major(), ver.Minor(), ver.Patch())
 }
 
-// Stream returns the stream of the version.
+// Stream returns the "release stream" of the version, i.e., the major and minor version without the patch version.
 func (v *Version) Stream() string {
 	ver := semver.MustParse(string(*v))
 	return fmt.Sprintf("v%d.%d", ver.Major(), ver.Minor())
 }
 
-// ReleaseBranch returns the release branch of the version.
+// ReleaseBranch returns the release branch which corresponds with this version.
 func (v *Version) ReleaseBranch(releaseBranchPrefix string) string {
-	branch := fmt.Sprintf("%s-%s", releaseBranchPrefix, v.Stream())
-	prerelese := semver.MustParse(string(*v)).Prerelease()
-	if prerelese != "" {
-		if strings.HasPrefix(prerelese, "1.") {
-			branch += "-1"
+	return fmt.Sprintf("%s-%s", releaseBranchPrefix, v.Stream())
+}
+
+// IsDevVersion returns true if the version includes the dev marker.
+// The dev marker can be used in a tag to indicate what the next release version will be. If present,
+// it means the version is of the form vX.Y.Z-<devTag>-<commitsSinceTag>-g<commitHash>, where vX.Y.Z
+// is the _next_ (currently unreleased) product version.
+func IsDevVersion(ver, devTag string) bool {
+	v := Version(ver)
+	pattern := fmt.Sprintf(`^v\d+\.\d+\.\d+(-\d+\.\d+)?-%s-\d+-g[0-9a-f]{12}(-dirty)?$`, devTag)
```

Comment:
Why should this return true if the version string doesn't include the `devTag`? I thought the entire purpose of this function was to tell if the dev tag was present.

That said, this function is currently unused so I think it can be removed?